R/fa.R

```diff
@@ -236,15 +238,16 @@ fa <- function(name,
 
 get_length_value_unit <- function(css_length) {
 
-  if (!grepl("^[0-9]+[a-z]+$", css_length)) {
+  if (!grepl("^[0-9\\.]+?[a-z]+$", css_length)) {
```

Comment:
I think the `?` makes the entire number portion optional, which isn't what we want. If you had a test like this, I think it would fail (because no error would be thrown):

```R
expect_error(get_length_value_unit("px"))

# Or, at a higher level:
expect_error(fa(name = "file", width = "px"))
```

I think the regex should be something like the one here:

```
testx <- function(x) {
  grepl("^[0-9]*\\.?[0-9]+[a-z]+$", x)
}

expect_true(testx("1px"))
expect_true(testx("1.0px"))
expect_true(testx(".1px"))

expect_false(testx("px"))
expect_false(testx(".px"))
expect_false(testx("1.px"))
```

Note that it doesn't count `1.px` to be valid.

Reply:
Thanks for this. I've adopted your regex and added a good number of unit tests.